src/service/fixed-layer.js

```diff
@@ -134,23 +200,29 @@ export class FixedLayer {
         dev().error(TAG, 'Aborting setup due to null stylesheet.');
         return;
       }
-      const {ownerNode} = stylesheet;
-      if (stylesheet.disabled ||
-              !ownerNode ||
-              ownerNode.tagName != 'STYLE' ||
-              ownerNode.hasAttribute('amp-boilerplate') ||
-              ownerNode.hasAttribute('amp-runtime') ||
-              ownerNode.hasAttribute('amp-extension')) {
+      const {cssRules, disabled, ownerNode} = stylesheet;
+      if (disabled ||
+          !ownerNode ||
+          ownerNode.tagName != 'STYLE' ||
+          ownerNode.hasAttribute('amp-boilerplate') ||
+          ownerNode.hasAttribute('amp-runtime') ||
+          ownerNode.hasAttribute('amp-extension')) {
         continue;
       }
-      this.discoverSelectors_(
-          stylesheet.cssRules, fixedSelectors, stickySelectors);
+      this.discoverSelectors_(cssRules, fixedSelectors, stickySelectors);
+    }
+    this.fixedSelectors_ = fixedSelectors;
+    this.stickySelectors_ = stickySelectors;
+
+    // Optimistically create the transfer layer. Otherwise, setVisible() may be
+    // called before the transfer layer is created, resulting in an incorrect
+    // initial visibility state. Alternatively, we could force creation of
+    // the transfer layer in setVisible().
+    if (this.fixedSelectors_.length > 0) {
```

Comment:
Delete, once enter/exit are delegated to the transfer layer.

Reply:
Done. Note this removes the optimization that avoids instantiating the transfer layer in some cases.